File: ios/VelaWallet/VelaWallet/Theme/VelaTheme.swift
import SwiftUI

// MARK: - Colors

enum VelaColor {
    static let bg = Color(hex: 0xFAFAF8)
    static let bgCard = Color.white
    static let bgWarm = Color(hex: 0xF5F3EF)

    static let textPrimary = Color(hex: 0x1A1A18)
    static let textSecondary = Color(hex: 0x7A776E)
    static let textTertiary = Color(hex: 0xB0ADA5)

    static let accent = Color(hex: 0xE8572A)
    static let accentSoft = Color(hex: 0xFFF0EB)

    static let green = Color(hex: 0x2D8E5F)
    static let greenSoft = Color(hex: 0xEDFAF2)

    static let blue = Color(hex: 0x4267F4)
    static let blueSoft = Color(hex: 0xEDF0FF)

    static let border = Color(hex: 0xECEAE4)

    // Token icon backgrounds
    static let ethBg = Color(hex: 0xEEF0F8)
    static let usdcBg = Color(hex: 0xEDF7F0)
    static let daiBg = Color(hex: 0xFFF8E7)

    // Network icon backgrounds
    static let arbBg = Color(hex: 0xE8F4FD)
    static let baseBg = Color(hex: 0xE8EEFF)
    static let opBg = Color(hex: 0xFFECEC)
}

// MARK: - Typography

enum VelaFont {
    static func heading(_ size: CGFloat) -> Font {
        .system(size: size, weight: .bold, design: .default)
    }

    static func title(_ size: CGFloat) -> Font {
        .system(size: size, weight: .semibold, design: .default)
    }

    static func body(_ size: CGFloat) -> Font {
        .system(size: size, weight: .regular, design: .default)
    }

    static func label(_ size: CGFloat) -> Font {
        .system(size: size, weight: .semibold, design: .default)
    }

    static func mono(_ size: CGFloat) -> Font {
        .system(size: size, weight: .medium, design: .monospaced)
    }

    static func caption() -> Font {
        .system(size: 12, weight: .medium, design: .default)
    }
}

// MARK: - Spacing & Radius

enum VelaRadius {
    static let card: CGFloat = 16
    static let cardSmall: CGFloat = 10
    static let full: CGFloat = 9999
    static let button: CGFloat = 16
}

enum VelaSpacing {
    static let screenH: CGFloat = 24
    static let cardPadding: CGFloat = 20
    static let itemGap: CGFloat = 14
}

// MARK: - Color Extension

extension Color {
    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: alpha
        )
    }
}

// MARK: - View Modifiers

struct VelaCardModifier: ViewModifier {
    func body(content: Content) -> some View {
        content
            .background(VelaColor.bgCard)
            .clipShape(RoundedRectangle(cornerRadius: VelaRadius.card))
            .overlay(
                RoundedRectangle(cornerRadius: VelaRadius.card)
                    .stroke(VelaColor.border, lineWidth: 1)
            )
    }
}

struct VelaPrimaryButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(VelaFont.label(16))
            .foregroundStyle(.white)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 17)
            .background(VelaColor.textPrimary)
            .clipShape(RoundedRectangle(cornerRadius: VelaRadius.button))
            .opacity(configuration.isPressed ? 0.85 : 1)
    }
}

struct VelaSecondaryButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(VelaFont.label(16))
            .foregroundStyle(VelaColor.textPrimary)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 17)
            .background(.clear)
            .clipShape(RoundedRectangle(cornerRadius: VelaRadius.button))
            .overlay(
                RoundedRectangle(cornerRadius: VelaRadius.button)
                    .stroke(VelaColor.border, lineWidth: 1.5)
            )
            .opacity(configuration.isPressed ? 0.7 : 1)
    }
}

struct VelaAccentButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(VelaFont.label(16))
            .foregroundStyle(.white)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 17)
            .background(VelaColor.accent)
            .clipShape(RoundedRectangle(cornerRadius: VelaRadius.button))
            .opacity(configuration.isPressed ? 0.85 : 1)
    }
}

extension View {
    func velaCard() -> some View {
        modifier(VelaCardModifier())
    }
}

File: android/VelaWallet/app/src/main/java/app/getvela/wallet/ui/theme/VelaTheme.kt
package app.getvela.wallet.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// MARK: - Colors (matches iOS VelaColor)

object VelaColor {
    val bg = Color(0xFFFAFAF8)
    val bgCard = Color.White
    val bgWarm = Color(0xFFF5F3EF)

    val textPrimary = Color(0xFF1A1A18)
    val textSecondary = Color(0xFF7A776E)
    val textTertiary = Color(0xFFB0ADA5)

    val accent = Color(0xFFE8572A)
    val accentSoft = Color(0xFFFFF0EB)

    val green = Color(0xFF2D8E5F)
    val greenSoft = Color(0xFFEDFAF2)

    val blue = Color(0xFF4267F4)
    val blueSoft = Color(0xFFEDF0FF)

    val border = Color(0xFFECEAE4)

    // Token icon backgrounds
    val ethBg = Color(0xFFEEF0F8)
    val usdcBg = Color(0xFFEDF7F0)
    val daiBg = Color(0xFFFFF8E7)

    // Network icon backgrounds
    val arbBg = Color(0xFFE8F4FD)
    val baseBg = Color(0xFFE8EEFF)
    val opBg = Color(0xFFFFECEC)
}

// MARK: - Typography (matches iOS VelaFont)

object VelaTypography {
    fun heading(sizeSp: Float) = TextStyle(
        fontSize = sizeSp.sp,
        fontWeight = FontWeight.Bold,
        fontFamily = FontFamily.Default,
    )

    fun title(sizeSp: Float) = TextStyle(
        fontSize = sizeSp.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = FontFamily.Default,
    )

    fun body(sizeSp: Float) = TextStyle(
        fontSize = sizeSp.sp,
        fontWeight = FontWeight.Normal,
        fontFamily = FontFamily.Default,
    )

    fun label(sizeSp: Float) = TextStyle(
        fontSize = sizeSp.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = FontFamily.Default,
    )

    fun mono(sizeSp: Float) = TextStyle(
        fontSize = sizeSp.sp,
        fontWeight = FontWeight.Medium,
        fontFamily = FontFamily.Monospace,
    )

    fun caption() = TextStyle(
        fontSize = 12.sp,
        fontWeight = FontWeight.Medium,
        fontFamily = FontFamily.Default,
    )
}

// MARK: - Spacing & Radius (matches iOS VelaSpacing / VelaRadius)

object VelaRadius {
    val card: Dp = 16.dp
    val cardSmall: Dp = 10.dp
    val button: Dp = 16.dp
}

object VelaSpacing {
    val screenH: Dp = 24.dp
    val cardPadding: Dp = 20.dp
    val itemGap: Dp = 14.dp
}

val VelaCardShape = RoundedCornerShape(VelaRadius.card)
val VelaButtonShape = RoundedCornerShape(VelaRadius.button)

// MARK: - Material 3 Theme Wrapper

private val VelaLightColorScheme = lightColorScheme(
    primary = VelaColor.textPrimary,
    secondary = VelaColor.accent,
    background = VelaColor.bg,
    surface = VelaColor.bgCard,
    onPrimary = Color.White,
    onSecondary = Color.White,
    onBackground = VelaColor.textPrimary,
    onSurface = VelaColor.textPrimary,
    outline = VelaColor.border,
)

@Composable
fun VelaWalletTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = VelaLightColorScheme,
        content = content,
    )
}
